### PR TITLE
Fix #27851 - Copy option gets the correct Public Link

### LINKED ISSUE #27851: Multiple Public Links: "Copy" option gets always the same link
### Steps to reproduce
1. Create several public links over the same file or folder
2. Use the "Copy" option to get each of them

### Expected behaviour
Each copied link is different 

### Current behaviour
Copied links are all the same (the first of the list)

### Server configuration
{"installed":"true","maintenance":"false","needsDbUpgrade":"false","version":"10.0.0.12","versionstring":"10.0.0","edition":"Enterprise","productname":"ownCloud"}

### PR DESCRIPTION
**Pull request by arunjohnkuruvilla**
## Description
Multiple Public Links: "Copy" option gets always the same link. This is because the template which adds a new row when a public link is created refers to the link by {{cid}} instead of {{id}}.

## Related Issue
https://github.com/owncloud/core/issues/27851

## Motivation and Context
Issue #27851 - Multiple Public Links: "Copy" option gets always the same link

## How Has This Been Tested?
Manually. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: core/js/sharedialoglinklistview.js [hunk 1]
```diff
@@ -20,8 +20,8 @@
 		'<li class="link-entry" data-id="{{id}}">' +
 			'<span class="link-entry--icon icon-public-white"></span>' +
 			'<span class="link-entry--title">{{linkTitle}}</span>' +
-			'<div class="minify"><input id="linkText-{{cid}}" class="linkText" type="text" readonly="readonly" value="{{link}}" /></div>' +
-			'<div class="link-entry--icon-button clipboardButton" data-clipboard-target="#linkText-{{cid}}">' +
+			'<div class="minify"><input id="linkText-{{id}}" class="linkText" type="text" readonly="readonly" value="{{link}}" /></div>' +
```
Comment (PVince81) [inline core/js/sharedialoglinklistview.js]: I suggest to use both `{{cid}}-{{id}}`.

cid guarantees that the element id is unique on the whole page.

Comment (arunjohnkuruvilla) [inline core/js/sharedialoglinklistview.js]: {{id}} is the id value for the particular share link returned from the backend. A share's id is kept unique by the backend. That should guarantee that the element id is unique for the whole application as well as the page.

Comment (PVince81) [inline core/js/sharedialoglinklistview.js]: What if there is another hidden sidebar that was rendered for another file list ? The sidebar might also contain the same elements with the same share ids but with a different view id.

This situation can usually happen if you navigate through the left sidebar elements "All files", "Shared with others" and open the sidebar for the same file/folder. When navigating away, the views aren't fully destroyed.

To keep it same I'd prefer using `{{cid}}-{{id}}`. There were already bugs in the past where checkboxes/labels didn't work due to the condition above.

**Comment by PVince81** [comment]
Would be good to get this finished until tomorrow as we'll build the RC version then.

Let me know if you can't and I'll take over. Thanks.

**Comment by arunjohnkuruvilla** [comment]
I have added the view identifier {{cid}} to the individual copy identifier. This should give a unique identifier every time.

**Comment by PVince81** [comment]
Awesome! Tested, works 👍

**Pull request merged**